account_avatax/security/avalara_salestax_security.xml

```diff
@@ -8,4 +8,10 @@
             name="domain_force"
         >['|',('company_id','=',False),('company_id','in',company_ids])]</field>
     </record>
+    <!--
+    company_id field was removed from Product Tax Codes,
+    and the corresponding record rule also.
+    This ensures the old record rule is removed from the database.
+    -->
+    <delete model="ir.rule" id="account_salestax_avatax_tax_codes_rule" />
```

Comment:
@dreispt 
You could use search instead of using id as using search will avoid deletion of record if External ID not found in the Database

`<delete model="ir.rule" search="[('id','=',ref('account_salestax_avatax_tax_codes_rule'))]"/>`

Currently, runbot is failing due to this reason
```
2020-09-25 11:34:17,135 158 INFO openerp_test odoo.modules.loading: loading account_avatax/security/avalara_salestax_security.xml
2020-09-25 11:34:17,149 158 WARNING openerp_test odoo.tools.convert: Skipping deletion for missing XML ID `'account_salestax_avatax_tax_codes_rule'`
Traceback (most recent call last):
File "/home/odoo/odoo-13.0/odoo/tools/cache.py", line 85, in lookup
r = d[key]
File "/home/odoo/odoo-13.0/odoo/tools/func.py", line 69, in wrapper
return func(self, *args, **kwargs)
File "/home/odoo/odoo-13.0/odoo/tools/lru.py", line 44, in __getitem__
a = self.d[obj].me
KeyError: ('ir.model.data', <function IrModelData.xmlid_lookup at 0x7f96bc2bc048>, 'account_avatax.account_salestax_avatax_tax_codes_rule')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/home/odoo/odoo-13.0/odoo/tools/convert.py", line 267, in _tag_delete
records += records.browse(self.id_get(d_id))
File "/home/odoo/odoo-13.0/odoo/tools/convert.py", line 657, in id_get
res = self.model_id_get(id_str, raise_if_not_found)
File "/home/odoo/odoo-13.0/odoo/tools/convert.py", line 663, in model_id_get
return self.env['ir.model.data'].xmlid_to_res_model_res_id(id_str, raise_if_not_found=raise_if_not_found)
File "/home/odoo/odoo-13.0/odoo/addons/base/models/ir_model.py", line 1681, in xmlid_to_res_model_res_id
return self.xmlid_lookup(xmlid)[1:3]
File "<decorator-gen-25>", line 2, in xmlid_lookup
File "/home/odoo/odoo-13.0/odoo/tools/cache.py", line 90, in lookup
value = d[key] = self.method(*args, **kwargs)
File "/home/odoo/odoo-13.0/odoo/addons/base/models/ir_model.py", line 1670, in xmlid_lookup
raise ValueError('External ID not found in the system: %s' % xmlid)
ValueError: External ID not found in the system: account_avatax.account_salestax_avatax_tax_codes_rule
2020-09-25 11:34:17,151 158 INFO openerp_test odoo.modules.loading: loading account_avatax/security/ir.model.access.csv
```

Reply:
same issue External ID not found with the search too, so instead tried to use <function> to call a method to delete if the record found

Reply:
I see your point, but the build is green...

Reply:
Looking better, there is not error but you see is an ugly warning.
Yes, it would be nice to avoid it.

Reply:
ok, If the build is successful but the ci/runbot says as failed.
![image](https://user-images.githubusercontent.com/4667326/94275871-1f4ad880-ff65-11ea-9565-af760a37851b.png)
I got confused about this.  Good to know

Reply:
The search ref does not work also.
Saw your code suggestion PR, but I think I have a simpler solution.
Hang on.

Reply:
Done and pushed.